mrpt_graphslam_2d/test/convertTest.cpp

```diff
@@ -31,3 +32,12 @@ TEST(ConvertTest, CObservation2DRangeScanTest)
 	EXPECT_EQ(obj.sensorPose,sensor_pose);
 }
 
+TEST(ConvertTest,CPose3DTest){
+	simxFloat position[3] ={1.00,1.00,1.00};
+	simxFloat quaternion[4] = {0.5,0.5,0.5,0.5};
+	CPose3D pose;
+	EXPECT_TRUE(convert(position,quaternion,pose));
+	float pitch = asin(-2*((quaternion[0]*quaternion[2])-(quaternion[1]*quaternion[3])));
+	EXPECT_EQ((double)pitch,pose.pitch());
```

Comment:
EXPECT_EQ() should not be used with floating numbers...
Please, take a look at: https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#floating-point-comparison